src/ExposureHistory/index.tsx

```diff
@@ -1,16 +1,30 @@
-import React, { FunctionComponent } from "react"
+import React, { FunctionComponent, useEffect } from "react"
 
 import { useExposureContext } from "../ExposureContext"
 import History from "./History"
 
 import { ExposureInfo, ExposureDatum } from "../exposure"
+import { useNavigation } from "@react-navigation/native"
 
 const toExposureList = (exposureInfo: ExposureInfo): ExposureDatum[] => {
   return exposureInfo
 }
 
 const ExposureHistoryScreen: FunctionComponent = () => {
-  const { lastExposureDetectionDate, exposureInfo } = useExposureContext()
+  const {
+    lastExposureDetectionDate,
+    exposureInfo,
+    refreshExposureInfo,
+  } = useExposureContext()
+
+  const navigation = useNavigation()
+
+  useEffect(() => {
+    // Refresh exposure info each time this screen gets focus
+    navigation.addListener("focus", () => {
```

Comment:
Nice one! I believe that we should return the result of that `addListener` inside a callback to avoid leaving unhandled promises laying around. We could return the same function that is the return value of the `addListener`:

``` typescript
{...}

useEffect(() => {
  const unsubscribe = navigation.addListener("focus", () => {
    refreshExposureInfo()
  })
  return unsubscribe
})

{...}
```

How do you feel about using the hook that `react-navigation` brings with it? That way we don't have to add this wider callback? Something like:

```typescript
import React, { FunctionCOmponent, useCallback } from "react"
import { useNavigation, useFocusEffect } from "@react-navigation/native"
{...}
  useFocusEffect(
    useCallback(() => {
      refreshExposureInfo()
    }, [refreshExposureInfo])
  );
{...}
```

Reply:
Not sure that would do but maybe something in that note? Maybe the docs on that hook help us more https://reactnavigation.org/docs/function-after-focusing-screen/

Reply:
Done!